### PR TITLE
Add opentelemetry tracing and export it to jaeger in dev setup

### PR DESCRIPTION
**Pull request by nsams**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -71,13 +71,14 @@ NEXT_PUBLIC_SITE_DOMAIN=main
...
-# apm, only used when API_ENABLE_APM
-KIBANA_PORT=5601
-APM_URL=http://localhost:8200
-
 # elastic
 ELASTIC_HOST=localhost
 ELASTIC_PORT=9200
 ELASTIC_USERNAME=
 ELASTIC_PASSWORD=
 ELASTIC_FULL_TEXT_SEARCH_INDEX=full-text-search
```
Comment (dkarnutsch) [inline .env:79]: not directly related to the PR, but these values are unused

File: demo/api/src/tracing.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    instrumentations: [
+        getNodeAutoInstrumentations({
+            "@opentelemetry/instrumentation-pg": {
+                requestHook: (span, queryInfo) => {
+                    span.updateName(`pg: ${queryInfo.query.text.substring(0, 100)}`);
+                },
+            },
+            "@opentelemetry/instrumentation-knex": {
+                enabled: false,
+            },
```
Comment (dkarnutsch) [inline demo/api/src/tracing.ts:21]:
```suggestion
```

or is it enabled by default?

File: docker-compose.yml [hunk 3]
```diff
@@ -34,6 +34,14 @@ services:
             AWS_ACCESS_KEY_ID: ${S3_ACCESS_KEY_ID}
             AWS_SECRET_ACCESS_KEY: ${S3_SECRET_ACCESS_KEY}
 
+    jaeger:
+        image: jaegertracing/all-in-one:1.42
```
Comment (dkarnutsch) [inline docker-compose.yml:38]:
```suggestion
        image: jaegertracing/all-in-one:1
```

if compatible, as we are also using postgres:12

**Review by dkarnutsch — CHANGES_REQUESTED** [review]
There are some changes which are not related to this PR, maybe there is a rebase required.

Other than that I think this looks fine and it is the way forward.

File: demo/api/src/tracing.ts (hunk 2 above)
Comment (nsams) [reply demo/api/src/tracing.ts:21]: it is enabled by default

File: .env (hunk 1 above)
Comment (nsams) [reply .env:79]: #990

**Comment by nsams** [comment]
> There are some changes which are not related to this PR, maybe there is a rebase required.
> 
> Other than that I think this looks fine and it is the way forward.

that's because it was a dependent PR, but the dependent is merged now (that's why i removed the comment)

somehow github doesn't recognize that (gitlab does)

**Review by dkarnutsch — APPROVED** [review]

File: demo/api/src/tracing.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            "@opentelemetry/instrumentation-graphql": {
+                mergeItems: true,
+            },
+            "@opentelemetry/instrumentation-nestjs-core": {
```
Comment (johnnyomair) [inline demo/api/src/tracing.ts:25]: We've decided that this instrumentation has no use for us, please remove

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Pull request merged**